set/set.go

```diff
@@ -0,0 +1,126 @@
+package set
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
+type Set[T comparable] struct {
+	values map[T]struct{}
+}
+
+func New[T comparable](_ T) *Set[T] {
```

Comment:
why `_ T` as argument?

Reply:
The argument is not required here, unless you want to init a set directly with some values, in that case it should accept a slice `[]T` and be put into the map